test/refactoring/ConvertAsync/convert_bool.swift

```diff
@@ -0,0 +1,271 @@
+  if !success && err != nil {
+    fatalError("oh no \(err!)")
+  }
+  print("not err")
+}
+
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 -I %S/Inputs -sdk %clang-importer-sdk-path  | %FileCheck -check-prefix=BOOL-WITH-ERR %s
+boolWithErr { success, err in
+  if err != nil && success == false {
+    fatalError("oh no \(err!)")
+  }
+  print("not err")
+}
+
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 -I %S/Inputs -sdk %clang-importer-sdk-path  | %FileCheck -check-prefix=BOOL-WITH-ERR %s
+boolWithErr { success, err in
+  if success == true && err == nil {
+  } else {
+    fatalError("oh no \(err!)")
+  }
+  print("not err")
+}
+
+// BOOL-WITH-ERR:      do {
+// BOOL-WITH-ERR-NEXT:   let success = try await boolWithErr()
+// BOOL-WITH-ERR-NEXT:   print("not err")
+// BOOL-WITH-ERR-NEXT: } catch let err {
+// BOOL-WITH-ERR-NEXT:   fatalError("oh no \(err)")
+// BOOL-WITH-ERR-NEXT: }
+
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 -I %S/Inputs -sdk %clang-importer-sdk-path  | %FileCheck -check-prefix=BOOL-WITH-ERR2 %s
+boolWithErr { success, err in
+  if success == true && err == nil {
+    print("hi")
+  } else {
+    fatalError("oh no \(err!)")
+  }
+  print("not err")
+}
+
+// BOOL-WITH-ERR2:      do {
+// BOOL-WITH-ERR2-NEXT:   let success = try await boolWithErr()
+// BOOL-WITH-ERR2-NEXT:   print("hi")
+// BOOL-WITH-ERR2-NEXT:   print("not err")
+// BOOL-WITH-ERR2-NEXT: } catch let err {
+// BOOL-WITH-ERR2-NEXT:   fatalError("oh no \(err)")
+// BOOL-WITH-ERR2-NEXT: }
+
+// Don't handle the below example as the force unwrap of err takes place under a different condition.
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 -I %S/Inputs -sdk %clang-importer-sdk-path  | %FileCheck -check-prefix=BOOL-DONT-HANDLE %s
+boolWithErr { success, err in
+  if !success {
+    if err != nil {
+      fatalError("oh no \(err!)")
+    }
+  }
+  if !success {
+    _ = err != nil ? fatalError("oh no \(err!)") : fatalError("some worries")
+  }
+  print("not err")
+}
+
+// BOOL-DONT-HANDLE:      let success = try await boolWithErr()
+// BOOL-DONT-HANDLE-NEXT: if !success {
+// BOOL-DONT-HANDLE-NEXT:   if <#err#> != nil {
+// BOOL-DONT-HANDLE-NEXT:     fatalError("oh no \(<#err#>!)")
+// BOOL-DONT-HANDLE-NEXT:   }
+// BOOL-DONT-HANDLE-NEXT: }
+// BOOL-DONT-HANDLE-NEXT: if !success {
+// BOOL-DONT-HANDLE-NEXT:   _ = <#err#> != nil ? fatalError("oh no \(<#err#>!)") : fatalError("some worries")
+// BOOL-DONT-HANDLE-NEXT: }
+// BOOL-DONT-HANDLE-NEXT: print("not err")
+
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 -I %S/Inputs -sdk %clang-importer-sdk-path  | %FileCheck -check-prefix=BOOL-DONT-HANDLE2 %s
+boolWithErr { success, err in
+  if !success {
+    func doThings() {
```

Comment:
See comment above - is there a good reason not to handle this case? Ie. if the force unwrap is used at all in the scope (nested or not), treat it as the error condition?

Reply:
The problem is that would count for things like this:

```
if !success {
  func doThings() {
    print(err!)
  }
  if err != nil {
    doThings()
  }
}
```

Where the `!success` condition doesn't appear to be relevant to the force unwrap. We could probably perform a more in-depth heuristic that also looks for other references to `err`, but I'm not sure whether or not that's worth it. What do you think?

Reply:
Makes sense, I wouldn't expect that case to be super common though. Another would be returning multiples bools and checking each both (nested). Though again, I assume that's uncommon. Erring on the side of leaving it and adding placeholders is probably okay for now though 👍